features/kafka/producer/src/main/java/org/opennms/features/kafka/producer/OpennmsKafkaProducer.java

```diff
@@ -408,6 +409,10 @@ private void processKafkaSendQueue() {
                     producer.send(producerRecord, (recordMetadata, e) -> {
                         if (e != null) {
                             LOG.warn("Failed to send record to producer: {}.", producerRecord, e);
+                            if(e instanceof TimeoutException) {
```

Comment:
Is it only `TimeoutException` in particular we want to retry for? Any other type of failure we are ok with dropping the message?

Reply:
Callback can get below retrievable Exceptions.

`     *                  CorruptRecordException
     *                  InvalidMetadataException
     *                  NotEnoughReplicasAfterAppendException
     *                  NotEnoughReplicasException
     *                  OffsetOutOfRangeException
     *                  TimeoutException
     *                  UnknownTopicOrPartitionException`
     
Out of these  we get `TimeoutException` when Kafka is offline for short period. That is the only case we want to handle here I think.

Reply:
Gotcha, makes sense!